src/sagemaker/jumpstart/estimator.py

```diff
@@ -0,0 +1,947 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+"""This module stores JumpStart implementation of Estimator class."""
+from __future__ import absolute_import
+import logging
+
+
+from typing import Dict, List, Optional, Union
+from sagemaker import session
+from sagemaker.async_inference.async_inference_config import AsyncInferenceConfig
+from sagemaker.base_deserializers import BaseDeserializer
+from sagemaker.base_serializers import BaseSerializer
+from sagemaker.debugger.debugger import DebuggerHookConfig, RuleBase, TensorBoardOutputConfig
+from sagemaker.debugger.profiler_config import ProfilerConfig
+
+from sagemaker.estimator import Estimator
+from sagemaker.explainer.explainer_config import ExplainerConfig
+from sagemaker.inputs import FileSystemInput, TrainingInput
+from sagemaker.instance_group import InstanceGroup
+
+from sagemaker.jumpstart.factory.estimator import get_deploy_kwargs, get_fit_kwargs, get_init_kwargs
+from sagemaker.jumpstart.factory.model import get_default_predictor
+from sagemaker.model_monitor.data_capture_config import DataCaptureConfig
+from sagemaker.predictor import PredictorBase
+
+
+from sagemaker.serverless.serverless_inference_config import ServerlessInferenceConfig
+from sagemaker.workflow.entities import PipelineVariable
+
+logger = logging.getLogger(__name__)
+
+
+class JumpStartEstimator(Estimator):
+    """JumpStartEstimator class.
+
+    This class sets defaults based on the model ID and version.
+    """
+
+    def __init__(
```

Comment:
This `__init__` is insane, I don't think I've ever seen this many params

Reply:
Isn't it? We are trying to provide intellisense for developers by exposing all the parameters of the parent class [Estimator](https://github.com/aws/sagemaker-python-sdk/blob/68885dcba8975f06f1e041dc7974c939e231e573/src/sagemaker/estimator.py#L2376). This forces us to expose all the parameters of the parent class constructor.

If we do not, and provide a catch all kwargs for some of these params, we would deteriorate the devX.